redis.conf

```diff
@@ -629,11 +629,9 @@ repl-diskless-sync-delay 5
 repl-diskless-sync-max-replicas 0
 
 # -----------------------------------------------------------------------------
-# WARNING: RDB diskless load is experimental. Since in this setup the replica
-# does not immediately store an RDB on disk, it may cause data loss during
-# failovers. RDB diskless load + Redis modules not handling I/O reads may also
```

Comment:
i thought of another reason for this text to be present:
> Since in this setup the replica does not immediately store an RDB on disk, it may cause data loss during failovers

this is this one: https://github.com/redis/redis/pull/6217

i.e. it's not about the master crashing while the replica performs full sync. it's about the replica not having a persistence file on the disk.

maybe that's what that text refers to?

Reply:
Think about the possible cases:
1. We care about durability and use AOF persistence: in this case it doesn't matter if diskless or disk-based replication, the dump.rdb created for replication will never be used again after replication finished. Are BGREWRITEAOF issued immediately after disk based replication in this case? If so, we should do that for diskless replication as well.
2. AOF and RDB disabled: in this case there will be a dump.rdb, but very quickly it will get old. We shouldn't even attempt to load this dump.rdb because we risk loading data that is months old. But we currently load it.
3. RDB only: here we have the chance of loading the dump.rdb created by replication, but remember that in diskless replication, from Redis 7 we started to increment `dirty` so it will likely in a few minutes trigger a BGSAVE even without other writes. In this case, we can only have a best effort to recover because we will likely load an old RDB every time there is a crash, either an RDB from replication or an RDB from BGSAVE, diskless or disk-based.

Number 3 is a case where we don't really mind about data loss because we use RDB. I think those that care about durability will use AOF.

So there's a lot of problems here and all scenarios have trade-offs.

Personally, I don't use Redis Cluster neither Sentinel so this problem is not existent. In my setup master uses AOF, replicas  RDB and we have diskless replication on both sides, working very well. Replica clients won´t perceive any down time for both crashes and restarts, and master down for a minute means just stale data. Overall there's no data loss even if disk is damaged (disk is synchronously replicated in 3 zones). I think this is also a very popular use case when you run Redis in Kubernetes, which will reschedule master asap if it crashes.

My point: a lot of setups are not affected by the fact that dump.rdb is not immediately stored after replication. This feature should not be considered experimental.
Even though I think we must explain the pros and cons clearly.

Reply:
I think that when discussing "data loss" in redis, we're discussing complete data loss, not just recent writes (since even a case of simple master crash will lose some writes not replicated yet. unless using AOF with every-write policy).
This is certainly the case when working with RDB snapshots, loading a year old rdb file, is technically not considered data loss in that perspective.

my point is that in case [3], when you set up a new replica, and maybe even promote it to master (for instance migrating redis to a different machine), there could be a case where is has absolutely no persistence and a complete data loss, which is not possible with disk-based replication.

I don't mind losing the experimental notice, many people are using it (even since redis 2.8), and it's certainly stable.
But let's try to keep the warning informative.

Reply:
@oranagra I've restored the information about data loss, and updated PR description.

Reply:
LGTM.
i noticed that the `on-empty-db` option says `this is safer` but it's not clear safer than what?
specifically since it is placed **before** the `swapdb` option.
maybe move it to after?

Reply:
Oh, of course, done